lib/client.js

```diff
@@ -298,6 +298,17 @@ MqttClient.prototype._setupStream = function () {
   // Echo connection errors
   parser.on('error', this.emit.bind(this, 'error'))
 
+  // auth
+  if (this.options.properties) {
+    if (!this.options.properties.authenticationMethod && this.options.properties.authenticationData) {
+      this.emit('error', new Error('Packet has no Authentication Method'))
```

Comment:
we might want to return after this? Emitting `'error'`  is a distructive action.

Reply:
I`ll add returning 'this' after emitting error